sdk/identity/Azure.Identity/src/IdentityCompatSwitches.cs

```diff
@@ -7,26 +7,37 @@ namespace Azure.Identity
 {
     internal class IdentityCompatSwitches
     {
-        private const string DisableInteractiveThreadpoolExecutionSwitchName = "Azure.Identity.DisableInteractiveBrowserThreadpoolExecution";
-        private const string DisableInteractiveThreadpoolExecutionEnvVar = "AZURE_IDENTITY_DISABLE_INTERACTIVEBROWSERTHREADPOOLEXECUTION";
+        internal const string DisableInteractiveThreadpoolExecutionSwitchName = "Azure.Identity.DisableInteractiveBrowserThreadpoolExecution";
+        internal const string DisableInteractiveThreadpoolExecutionEnvVar = "AZURE_IDENTITY_DISABLE_INTERACTIVEBROWSERTHREADPOOLEXECUTION";
+        internal const string DisableCP1ExecutionSwitchName = "AZURE.IDENTITY.DISABLECP1";
+        internal const string DisableCP1ExecutionEnvVar = "AZURE_IDENTITY_DISABLE_CP1";
 
         public static bool DisableInteractiveBrowserThreadpoolExecution
+            => GetConfigValue(DisableInteractiveThreadpoolExecutionSwitchName, DisableInteractiveThreadpoolExecutionEnvVar);
+
+        public static bool DisableCAE
+            => GetConfigValue(DisableCP1ExecutionSwitchName, DisableCP1ExecutionEnvVar);
+
+        /// <summary>
+        /// Determines if either an AppContext switch or its corresponding Environment Variable is set
+        /// </summary>
+        /// <param name="switchName">Name of the AppContext switch.</param>
+        /// <param name="envName">Name of the Environment variable.</param>
+        /// <returns>true if either the switch or the environment variable is set to true. False if both are set to false or unset.</returns>
+        private static bool GetConfigValue(string switchName, string envName)
         {
-            get
+            if (!AppContext.TryGetSwitch(switchName, out bool ret) || ret == false)
```

Comment:
@schaabs I made this tweak to the logic for a couple reasons
- If either the environment variable or the AppContext switch is true, the setting will take effect. This means that the AppContext switch can now be "unset" even after it has been set.
- Tests can now validate the effect of this setting without concern for ordering or previous state

Reply:
@pakrym what are your thoughts on this. Originally I followed the same logic you had for compat switches in Azure.Core, but this does make it a bit more testable. The only issue would be if users expected explicitly setting the `AppContext` switch to `false` would take precedence over the environment variable being set.

Reply:
My number one concern is that all appswitches behave the same. It would be great if we were consistent with the way .NET does it (https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionSettings.cs#L135-L158)

Reply:
I think consistency makes sense. I just don't like that you can't "unset" an AppConfig.

Reply:
I think you can if you set it via SetData https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/AppContext.cs#L105

Reply:
Might be good to move switch logic into core\shared and add a utility type to TestFramework to set-unset them.

Reply:
Hmm - the [docs ](https://docs.microsoft.com/en-us/dotnet/api/system.appcontext?view=net-5.0#methods)don't mention `SetData` but I see it's there.

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/AppContext.cs#L48

Reply:
It might be .NET core only. I'll be fine doing private reflection in the helper type to clear the switch.